### PR TITLE
System-wide installation improvements

### PR DESCRIPTION
**Pull request by FedericoCeratto**
Add contributing guide
Add Debian packaging, initial version
Prepare NimHA for system-wide installation
Handle system-wide paths
Setup Nimble in dedicated dir and install build deps
Code cleanup

### CODE REVIEW

File: nimha.nim [hunk 1]
```diff
@@ -184,70 +212,44 @@ proc compileIt() =
...
-      quit()
-    else:
-      echo "outputXiaomiListener compiling done\n\n"
+    compileModule(devC, "nimha_xiaomilistener")
 
 
 proc requirements() =
-  discard existsOrCreateDir(replace(getAppDir(), "/nimhapkg/mainmodules", "") & "/tmp")
   when defined(dev):
+    discard existsOrCreateDir(getTmpDir())
```
Comment (ThomasTJdev) [inline nimha.nim:252]: `getTmpDir()` requires a param:
```
proc getTmpDir*(modulename: string): string =
```
==>
```
proc getTmpDir*(modulename = ""): string =
```

**Review by ThomasTJdev — CHANGES_REQUESTED** [review]
Sorry for the slow review.

All of the `-d:release` marked #installpath needs to manually created. How do you plan to do this? A manually check/copy at init when running in `release`?
The master branch needs to be automated. I can create a devel-branch we can work on, where the automation process for the lazy user can be implemented at the end?

**Comment by ThomasTJdev** [comment]
> Sorry for the slow review.
> 
> All of the `-d:release` marked #installpath needs to manually created. How do you plan to do this? A manually check/copy at init when running in `release`?
> The master branch needs to be automated. I can create a devel-branch we can work on, where the automation process for the lazy user can be implemented at the end?

Ohh give me a sec. I didn't checkout the debian files.

**Comment by FedericoCeratto** [comment]
Creating system users, groups, directories, config files is left to the installer. apt, zypper, and a (yet to be written) install.sh do that each in its own way.

**Comment by ThomasTJdev** [comment]
I'm running Arch which made it a little difficult to test the `-d:release` mode. I tried to combine some bash for a install script - [gist link](https://gist.github.com/ThomasTJdev/898e9de422d5a1c4f33f9b546b46e6b0).

I was able to commit directly to you `devel` branch, but only by pushing to your repo.

I'll merge now, but I'll merge to a devel branch. We'll merge the devel with master, when `-d:release` works for main linux distros - ok?

**Review by ThomasTJdev — APPROVED** [review]

**Pull request merged**